File: frontend/src/types/Inventory.ts
export interface Inventory {
  inventory_id: number;
  product_id: number;
  current_stock: number;
  reorder_level: number;
  last_restock_date?: Date;
}

export interface InventoryWithProduct extends Inventory {
  product_name?: string;
  model?: string;
  price?: number;
  isLowStock?: boolean;
}

export interface InventoryCreate extends Omit<Inventory, 'inventory_id'> {
  inventory_id?: number;
}

export interface InventoryUpdate extends InventoryCreate {
  inventory_id: number;
}

export interface StockUpdate {
  current_stock: number;
}

export interface LowStockItem extends InventoryWithProduct {
  product_name: string;
  price: number;
}

File: backend/internal/models/inventory.go
package models

import (
	"time"
)

// Inventory tracks stock levels
type Inventory struct {
	InventoryID     int        `db:"inventory_id" json:"inventory_id"`
	ProductID       int        `db:"product_id" json:"product_id"`
	CurrentStock    int        `db:"current_stock" json:"current_stock"`
	ReorderLevel    int        `db:"reorder_level" json:"reorder_level"`
	LastRestockDate *time.Time `db:"last_restock_date" json:"last_restock_date,omitempty"`
} 